packages/file-search/src/node/file-search-service-impl.ts

```diff
@@ -110,14 +110,7 @@ export class FileSearchServiceImpl implements FileSearchService {
         const args = ['--files', '--case-sensitive'];
         if (options.includePatterns) {
             for (const includePattern of options.includePatterns) {
-                let includeGlob = includePattern;
-                if (!includeGlob.endsWith('*')) {
-                    includeGlob = `${includeGlob}*`;
-                }
-                if (!includeGlob.startsWith('*')) {
-                    includeGlob = `*${includeGlob}`;
-                }
-                args.push('--glob', includeGlob);
+                args.push('--glob', includePattern);
```

Comment:
should we do this to the ignored patterns too ? line 122 - 131

Reply:
good question, for vscode APIs i believe so

They were introduced for quick file though. Do you remember for which use cases?

Reply:
the ignored pattern was there long time ago with the default value of `.git`. so i guess it was used to skip files in the `.git` folder. 
maybe we should leave line 122 - 131 as is, at least for now, as we have no idea if clients can possibly pass string literals (instead of globs) with `option.defaultIgnorePatterns`.

Reply:
i will introduce a new option for vscode extensions, e.g. `excludPatterns` and deprecate `defaultIgnorePatterns`. Magic around `defaultIgnorePatterns` should go to the quick file client.

Reply:
I think we misdesigned search APIs again :) `includePatterns` and `excludePatterns` should per a root URI, not for all root URIs

By default VS Code extensions expect that settings like `files.exclude` and `search.exclude` are applied and it can be different for each root.

It's actually the same for the quick file in VS Code, so you cannot search in `node_modules` without adjusting `search.exclude` folder. I like how it works now in Theia that you can toggle it. Should we align it as in VS Code? cc @svenefftinge

Reply:
I did not change how the quick file works, but improved APIs to support vscode APIs.